AppBuilder/platform/views/ABViewReportsManager.js

```diff
@@ -680,9 +774,24 @@ module.exports = class ABViewReportsManager extends ABViewReportsManagerCore {
                                     break;
                                  case "number":
                                     reportRow[col] = parseFloat(
-                                       reportRow[col] || 0
+                                       (reportRow[col] || 0)
+                                          .toString()
+                                          .replace(/[^\d.-]/g, "")
                                     );
                                     break;
+                                 case "date":
+                                 case "datetime":
+                                    reportRow[col] = row[columnName];
+                                    if (reportRow[col]) {
+                                       if (!(reportRow[col] instanceof Date)) {
+                                          reportRow[col] = new Date(
```

Comment:
should use one of the `.toDate()` ,  `.toDateFormat()` or other ABFactory methods instead of referencing `moment` here directly.